pkg/generator/generate.go

```diff
@@ -232,10 +232,10 @@ func (g *Generator) identifierFromFileName(fileName string) string {
 	s := filepath.Base(fileName)
 	for _, ext := range g.config.ResolveExtensions {
 		trimmed := strings.TrimSuffix(s, ext)
-		if trimmed == s {
+		if trimmed != s {
+			s = trimmed
```

Comment:
The previous version was causing the loop to always break, because trimming `.json` from a YAML filename didn't cause any changes. The loop would never get to the point where the `.yaml` extension was actually trimmed from the filename.